src/Microsoft.DotNet.TemplateLocator/TemplateLocator.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using NuGet.Common;
+using NuGet.Protocol;
+
+namespace Microsoft.DotNet.TemplateLocator
+{
+    public sealed class TemplateLocator
+    {
+        private DirectoryInfo _dotnetSdkTemplatesLocation;
+
+        public TemplateLocator()
+        {
+            string mockTemplateLocation = Environment.GetEnvironmentVariable("MOCKDOTNETSDKTEMPLATESLOCATION");
+            if (!string.IsNullOrWhiteSpace(mockTemplateLocation))
+            {
+                _dotnetSdkTemplatesLocation = new DirectoryInfo(mockTemplateLocation);
+            }
+        }
+
+        public IReadOnlyCollection<IOptionalSdkTemplatePackageInfo> GetDotnetSdkTemplatePackages(string sdkVersion)
+        {
+            if (_dotnetSdkTemplatesLocation == null)
+            {
+                return Array.Empty<IOptionalSdkTemplatePackageInfo>();
+            }
+
+            return LocalFolderUtility
+                .GetPackagesV2(_dotnetSdkTemplatesLocation.FullName, new NullLogger())
```

Comment:
Can GetPackagesV2 return null?

Reply:
Hmm actually it would. https://github.com/NuGet/NuGet.Client/blob/784274cc9b2002ca73055e8134064261118d8a01/src/NuGet.Core/NuGet.Protocol/Utility/LocalFolderUtility.cs#L194

I will throw an exception when it is null

Reply:
Just made it return empty array when null